libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
It is not tight to a specific application, it is only tight to an application. I don't care in my component if I have a JApplicationUnitTest or JApplicationCMS or JApplicationCLI. I need stuff like $app->getContainer() or $app->getConfig(). If the container is becomes the master holder, then we don't need an application at all.